src/main/java/net/dv8tion/jda/api/entities/MessageChannel.java

```diff
@@ -1565,7 +1565,68 @@ default MessagePaginationAction getIterableHistory()
+     *     {
+     *         if (!history.isEmpty())
+     *         {
+     *             Message firstMsg = history.getRetrievedHistory().get(0);
+     *             channel.sendMessage(firstMsg).queue();
+     *         }
+     *         else
+     *             channel.sendMessage("No history for this channel!").queue();
+     *     });
+     * }
+     * </code></pre>
+     *
+     * <p>The following {@link net.dv8tion.jda.api.requests.ErrorResponse ErrorResponses} are possible:
+     * <ul>
+     *     <li>{@link net.dv8tion.jda.api.requests.ErrorResponse#MISSING_ACCESS MISSING_ACCESS}
+     *     <br>The request was attempted after the account lost access to the {@link net.dv8tion.jda.api.entities.Guild Guild}
+     *         typically due to being kicked or removed, or after {@link net.dv8tion.jda.api.Permission#MESSAGE_READ Permission.MESSAGE_READ}
+     *         was revoked in the {@link net.dv8tion.jda.api.entities.TextChannel TextChannel}</li>
+     *
+     *     <li>{@link net.dv8tion.jda.api.requests.ErrorResponse#MISSING_PERMISSIONS MISSING_PERMISSIONS}
+     *     <br>The request was attempted after the account lost
+     *         {@link net.dv8tion.jda.api.Permission#MESSAGE_HISTORY Permission.MESSAGE_HISTORY} in the
+     *         {@link net.dv8tion.jda.api.entities.TextChannel TextChannel}.</li>
+     *
+     *     <li>{@link net.dv8tion.jda.api.requests.ErrorResponse#UNKNOWN_CHANNEL UNKNOWN_CHANNEL}
+     *     <br>The request was attempted after the channel was deleted.</li>
+     * </ul>
+     *
+     * @param  limit
+     *         The amount of messages to be retrieved. Minimum: 1, Max: 100.
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         Provided {@code limit} is less than {@code 1} or greater than {@code 100}.
+     * @throws net.dv8tion.jda.api.exceptions.InsufficientPermissionException
+     *         If this is a {@link net.dv8tion.jda.api.entities.TextChannel TextChannel} and the logged in account does not have
+     *         <ul>
+     *             <li>{@link net.dv8tion.jda.api.Permission#MESSAGE_READ Permission.MESSAGE_READ}</li>
+     *             <li>{@link net.dv8tion.jda.api.Permission#MESSAGE_HISTORY Permission.MESSAGE_HISTORY}</li>
+     *         </ul>
+     *
+     * @return {@link net.dv8tion.jda.api.entities.MessageHistory.MessageRetrieveAction MessageHistory.MessageRetrieveAction}
+     *         <br>Provides a {@link net.dv8tion.jda.api.entities.MessageHistory MessageHistory} object with messages - beginning from the start - loaded into it.
+     *
+     * @see    net.dv8tion.jda.api.entities.MessageChannel#getHistoryAfter(long, String) MessageChannel.getHistoryAfter(long, String)
+     * @see    net.dv8tion.jda.api.entities.MessageHistory#getHistoryAfter(MessageChannel, String) MessageHistory.getHistoryAfter(MessageChannel, String)
+     */
+    @CheckReturnValue
+    default MessageHistory.MessageRetrieveAction getHistoryFromBeginning(int limit)
```

Comment:
There should be a `@Nonnull` annotation from `javax.annotations` here as well for the return value. This is a pattern we are starting to deploy in another PR and it should already be done here as well. (only for the new method to avoid conflicts)

Reply:
Not sure if I understand you correctly, do you mean that:
```suggestion
    @CheckReturnValue
    @NonNull
    default MessageHistory.MessageRetrieveAction getHistoryFromBeginning(int limit)
```
?

Never worked with the NonNull annotation, I just know that attributes / parameters sometimes have them.

Reply:
Sort them by length for better readability.

```java
@Nonnull
@CheckReturnValue
default MessageHistory.MessageRetrieveAction getHistoryFromBeginning(int limit)
{
....
```

Reply:
Note: It isn't `@NotNull`. (which is from Jetbrains). It is `@Nonnull` which is from `javax.annotations`

Reply:
@MinnDevelopment You're right, did it like you suggested it.
@DV8FromTheWorld Thanks, confused those two.